flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/IcebergTableSink.java

```diff
@@ -71,7 +72,10 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
         "Unbounded data stream doesn't support overwrite operation.");
 
     List<String> equalityColumns =
-        tableSchema.getPrimaryKey().map(UniqueConstraint::getColumns).orElseGet(ImmutableList::of);
+        tableSchema
+            .getPrimaryKey()
+            .map(UniqueConstraint::getColumns)
+            .orElseGet(() -> getEqualityColumnsFromProps(writeProps));
```

Comment:
We don't parse options this way.

You should refer to FlinkWriteOptions. Put`equlality columns` in `writeOptions`( `FlinkSink.Builder#equalityFieldColumns`) like we did in `FlinkSink.Builder#upsert`.